File: Backend/app/models/user_model.py
from datetime import datetime
from pydantic import BaseModel, EmailStr, constr
from typing import Optional
from enum import Enum


class UserTypeEnum(str, Enum):
    owner = 'owner'
    customer = 'customer'
    admin = 'admin'


class User(BaseModel):
    FirstName: str
    LastName: str
    MiddleName: str
    PhoneNumber: str  # constr(pattern=r'^\(\d{3}\)-\d{3}-\d{4}$')
    Email: EmailStr
    Id: int
    Created: Optional[datetime] = None
    LastUpdated: Optional[datetime] = None
    LastLogged: Optional[datetime] = None
    Password: str
    UserType: UserTypeEnum

File: Frontend/UNCGEats/src/app/models/user.model.ts
export interface User {
  FirstName: string;
  LastName: string;
  MiddleName: string;
  PhoneNumber: number;
  Email: string;
  Id?: number;
  Created?: Date;
  LastUpdated?: Date;
  LastLogged?: Date;
  Password: string;
  UserType: string;
  deleted?: boolean
}
